modules/spring-config/src/main/java/org/mule/runtime/config/internal/context/lazy/ComponentInitializationRequest.java

```diff
@@ -69,11 +70,13 @@ public static class Builder {
      * @param keepPrevious                    Whether previously initialized components should be kept unchanged.
      */
     public Builder(ArtifactAstDependencyGraph graph,
+                   ArtifactAstDependencyGraph graphWithoutMacroExpansion,
```

Comment:
is the macroexpanded graph required for anything?

Reply:
Graph without macro expansion was added for the validation with lazy init. 
_Graph_(the macroexpanded graph) has been used/ and is still being used when we get minimalAst to update the requestedLocations during the creation of components.